include/ginkgo/core/base/batch_lin_op.hpp

```diff
@@ -671,9 +671,12 @@ class BatchDense;
  *
  * @see BatchScalable
  */
-template <typename ValueType>
 class EnableBatchScaling : public BatchScalable {
```

Comment:
Again, this is because `EnableBatchSolver` cannot depend on the value type directly. I will eventually remove `EnableBatchScaling` completely and introduce `BatchIdentity` and `BatchDiagonal`, and "scaling" (a misnomer to begin with, not sure why we called it that) will be taken care of with their `apply` functions.